src/database/DataTypes/TransactionItem.js

```diff
@@ -72,7 +72,10 @@ export class TransactionItem extends Realm.Object {
    * are finalised, whereas customer invoices immediately influence stock levels.
    * @param {double} quantity The total quantity to set across all batches
    */
-  setTotalQuantity(database, quantity) {
+   // TODO: document function param
+  setTotalQuantity(database, requestedQuantity, referenceQuantity = -1) {
+    let quantity = requestedQuantity;
+    if (referenceQuantity > 0) quantity = Math.min(requestedQuantity, referenceQuantity);
```

Comment:
My comment doesn't apply here @andreievg, because if it is called when the transaction is finalised, it won't cause any obvious error. It's a very different situation to the one I commented on.